types/mergerino/index.d.ts

```diff
@@ -26,7 +26,7 @@ export type DeletePatch = undefined;
  * and the merge function as the second. The return value will be the replacement.
  * The value you return will bypass merging logic and simply overwrite the property.
  */
-export type FunctionPatch<T> = (val: T, merge: Merge<T extends object ? T : never>) => T;
+export type FunctionPatch<T> = (val: Exclude<T, undefined>, merge: Merge<T extends object ? T : never>) => T;
```

Comment:
I think that it is wrong, because you can patch an optional property using function patch and in this case the value will ```undefined```.

For example:
```typescript
interface Foo {
    bar: number,
    baz?: number
}

const item: Foo = {
    bar: 1
};

const out = merge(item, { baz: (bazValue) => 42 });
```

It is correct mergerino code and in this example ```bazValue``` is ```undefined```.
We need to keep it as is, without exclude ```undefined```.